core/src/main/java/apoc/export/graphml/XmlGraphMLReader.java

```diff
@@ -336,6 +336,7 @@ public long parseXML(Reader input, TerminationGuard terminationGuard) throws XML
             tx.rollback();
             throw e;
         } finally {
+            reader.close();
```

Comment:
If this call throws an exception we will not call `tx.close()`. Might not be an issue in this case though, if I remember correctly transactions are closed on commit/rollback anyway.